### ISSUE TITLE
`migrate` returns 1 when migration already done

### ISSUE THREAD

**Issue by smndtrl** [openfga/openfga#1832]
### Checklist

- [X] I have looked into the [README](https://github.com/openfga/openfga/blob/main/README.md) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://openfga.dev/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/openfga/openfga/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/openfga/openfga/releases/latest) of OpenFGA and the issue still persists.
- [X] I have searched the [OpenFGA Community](https://openfga.dev/community) and have not found a suitable solution or answer.
- [X] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

### Description

As part of our deployments we run a `migrate` every time before we start a `run` container. Now we discovered that, because we had to re-deploy the same version of openfga twice, the second time the migration errors.

`Error: failed to get db version: ERROR: relation "goose_db_version" already exists (SQLSTATE 42P07)`

We assume this might be unintended and something like a return `0` with a no migration necessary log should have happened.

### Expectation

When migration is already applied return code would be `0`

### Reproduction

1. Run migrate
2. Run migrate

### Store data

```yaml
model_file: |
  model
    schema 1.1
  type user
  type organization
    relations
      define member: [user]
tuples:
  - user: user:anne
    relation: member
    object: organization:openfga
  - user: user:bob
    relation: member
    object: organization:openfga
tests: # remove this if not a bug in Check or ListObjects API
  - name: test-1
    check:
      - user: user:anne
        object: organization:openfga
        assertions:
          member: true
    list_objects:
      - user: user:anne
        type: organization
        assertions:
          member:
            - organization:openfga
```

### OpenFGA version

latest

### How are you running OpenFGA?

In Docker

### What datastore are you using?

Postgres

### OpenFGA Flags

-

**Comment by miparnisari** [comment]
Hey @smndtrl, I am trying to reproduce this issue locally and I cannot:

```shell
[9/10/24 7:42:59] ~/GitHub/openfga-main (main) $ ./dist/openfga migrate --datastore-engine postgres --datastore-uri 'postgres://postgres:password@localhost:5432/postgres'
2024/10/09 19:43:33 current version 5
2024/10/09 19:43:33 running all migrations
2024/10/09 19:43:33 migration done
[9/10/24 7:43:33] ~/GitHub/openfga-main (main) $ echo $?
0
[9/10/24 7:43:36] ~/GitHub/openfga-main (main) $ ./dist/openfga migrate --datastore-engine postgres --datastore-uri 'postgres://postgres:password@localhost:5432/postgres'
2024/10/09 19:43:37 current version 5
2024/10/09 19:43:37 running all migrations
2024/10/09 19:43:37 migration done
[9/10/24 7:43:37] ~/GitHub/openfga-main (main) $ echo $?
0
[9/10/24 7:43:38] ~/GitHub/openfga-main (main) $ 
```

**Comment by renan-abm** [comment]
Related to https://github.com/openfga/openfga/issues/479.

Hey, I just hit this same issue. It happens if you are using PgBouncer in transaction mode. Using the default `session` mode works fine but negates almost all the benefits of using PgBouncer.

Here's a minimum reproduction I made using the compose file in this repository and adding PgBouncer. The key change here was setting `PGBOUNCER_POOL_MODE` to transaction.

```yaml
services:
  postgres:
    image: postgres:14
    container_name: postgres
    command: postgres -c 'max_connections=100'
    networks:
      - default
    ports:
      - '5432:5432'
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=password
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U postgres']
      interval: 5s
      timeout: 5s
      retries: 5

  pgbouncer:
    image: bitnami/pgbouncer:latest
    container_name: pgbouncer
    environment:
      - POSTGRESQL_HOST=postgres
      - POSTGRESQL_USERNAME=postgres
      - POSTGRESQL_PASSWORD=password
      - PGBOUNCER_POOL_MODE=transaction
    ports:
      - '6432:6432'
    depends_on:
      postgres:
        condition: service_healthy
    networks:
      - default
    healthcheck:
      test:
        [
          'CMD-SHELL',
          'PGPASSWORD=password psql --host=localhost --port=6432 --user=postgres --echo-errors --command="SHOW USERS" pgbouncer > /dev/null'
        ]
      interval: 5s
      timeout: 5s
      retries: 5

  migrate:
    depends_on:
      pgbouncer:
        condition: service_healthy
    image: openfga/openfga:latest
    container_name: migrate
    environment:
      - OPENFGA_DATASTORE_ENGINE=postgres
      - OPENFGA_DATASTORE_URI=postgres://postgres:password@pgbouncer:6432/postgres?sslmode=disable
    command: migrate
    networks:
      - default

  openfga:
    depends_on:
      migrate:
        condition: service_completed_successfully
    image: openfga/openfga:latest
    container_name: openfga
    command: run
    environment:
      - OPENFGA_DATASTORE_ENGINE=postgres
      - OPENFGA_DATASTORE_URI=postgres://postgres:password@pgbouncer:6432/postgres?sslmode=disable
      - OPENFGA_DATASTORE_MAX_OPEN_CONNS=100 #see postgres container
      - OPENFGA_PLAYGROUND_ENABLED=true
    networks:
      - default
    ports:
      - '8080:8080' #http
      - '8081:8081' #grpc
      - '3000:3000' #playground
      - '2112:2112' #prometheus metrics
    healthcheck:
      test: ['CMD', '/usr/local/bin/grpc_health_probe', '-addr=openfga:8081']
      interval: 5s
      timeout: 30s
      retries: 3
```

**Comment by makhil006** [comment]
+1

It works on 5432 but not on 6543. We are using OpenFGA on k8s and Supabase for Postgres.
